src/main/java/org/mifos/connector/gsmastub/api/ApiOriginFilter.java

```diff
@@ -1,14 +1,67 @@
 package org.mifos.connector.gsmastub.api;
 
-import javax.servlet.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
+import java.util.Enumeration;
+import java.util.Iterator;
+import java.util.Optional;
 
 @javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.SpringCodegen", date = "2022-09-27T11:46:46.417Z[GMT]")
+@Component
 public class ApiOriginFilter implements Filter {
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private static final String[] HEADERS_TO_TRY = {
+            "X-Forwarded-For",
+            "Proxy-Client-IP",
+            "WL-Proxy-Client-IP",
+            "HTTP_X_FORWARDED_FOR",
+            "HTTP_X_FORWARDED",
+            "HTTP_X_CLUSTER_CLIENT_IP",
+            "HTTP_CLIENT_IP",
+            "HTTP_FORWARDED_FOR",
+            "HTTP_FORWARDED",
+            "HTTP_VIA",
+            "REMOTE_ADDR" };
+    private String getClientIpAddress(HttpServletRequest request) {
+        logger.info("Trying to log headers");
+        for (String header : HEADERS_TO_TRY) {
+            String ip = request.getHeader(header);
+            if (ip != null && ip.length() != 0 && !"unknown".equalsIgnoreCase(ip)) {
+                logger.info(ip);
+            }
+        }
+
+        return request.getRemoteAddr();
+    }
     @Override
     public void doFilter(ServletRequest request, ServletResponse response,
-            FilterChain chain) throws IOException, ServletException {
+                         FilterChain chain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) request;
+        getClientIpAddress(req);
+//        Enumeration<String> stringEnum=req.getAttributeNames();
+//        Iterator<String>it=stringEnum.asIterator();
+//        while(it.hasNext()){
+//            logger.info("Attribute:{}",it.next());
+//        }
+//        String clientIP = req.getHeader("X-Forwarded-For");
+        String tenant = req.getHeader("Platform-TenantId");
+        Optional<String> tenantVal = Optional.ofNullable(tenant);
+        if (tenantVal.isPresent()) {
+            logger.info("Tenant Name is : {}", tenant);
```

Comment:
This looks like an unnecessary wrapping. what is the use of creating the optional wrapper, when you already have the object? try methods like `nonNull(foo)`.  logging empty tenant is also fine IMO